### PR TITLE
Update DDR4 OC Guide.md

### PR DESCRIPTION
**Pull request by LunarPSD**
Many grammar and text corrections as well as updated information, links, and easier readability.

### CODE REVIEW

File: DDR4 OC Guide.md [hunk 1]
```diff
@@ -37,24 +37,21 @@
...
+* I wouldn't recommend the AIDA64 memory test and the [Memtest64](https://forums.anandtech.com/threads/techpowerups-memtest-64-is-it-better-than-hci-memtest-for-determining-stability.2532209/) as they are both not very good at finding memory errors.
 ### Recommended
 * [TM5](http://testmem.tz.ru/tm5.rar) with any of the configs listed:
   * [Extreme config by anta777](https://drive.google.com/file/d/1uegPn9ZuUoWxOssCP4PjMjGW9eC_1VJA) (recommended). Make sure to load the config. It should say 'Customize: Extreme1 @anta777' if loaded.  
   Credits: [u/nucl3arlion](https://www.reddit.com/r/overclocking/comments/dlghvs/micron_reve_high_training_voltage_requirement/f4zcs04/).
-  * [Here](https://www.overclock.net/threads/memory-testing-with-testmem5-tm5-with-custom-configs.1751608/) is a link to TM5 pre-packaged with many configs.
+  * [Here](https://drive.google.com/drive/u/0/folders/1E-5ZsrlKwr1SlEWxK3iLebms2k3EkQc) is a link to TM5 pre-packaged with many configs.
```
Comment (integralfx) [inline DDR4 OC Guide.md]: Why did you change the URL? It doesn't seem to work anyway

File: DDR4 OC Guide.md [hunk 2]
```diff
@@ -37,24 +37,21 @@
...
+  * If you experience issues with all threads crashing upon launch with the extreme config it might help to edit the row "Testing Window Size (Mb)=1408". Replace the window size with your total RAM (minus some margin for Windows) divided by your processor's available threads (e.g. 12800/16 = 800 MB per thread).
 * [OCCT](https://www.ocbase.com/) with the dedicated memory test using either SSE or AVX instructions.
   * Note that AVX and SSE can vary in error detection speed. On Intel-based systems, SSE appears better for testing IMC voltages while AVX appears better for DRAM voltage.
-  * The Large AVX2 CPU test is a great stability test for your CPU and RAM at the same time. The more you tune your ram the harder it'll be to stable in this test.
+  * The Large AVX2 CPU test is a great stability test for your CPU and RAM at the same time. The more you tune your ram the harder it'll be to stable in this test. Be sure to run Normal Mode as Extreme will not use as much RAM.
+  * The VRAM test at maximum utilization in conjunction with Prime95 Large FFTs will stress FCLK and is recommended when testing FCLK stability.
```
Comment (integralfx) [inline DDR4 OC Guide.md:55]: Has anyone else verified this?

File: DDR4 OC Guide.md [hunk 3]
```diff
@@ -94,39 +91,46 @@ You should always test with a variety of stress tests to ensure your overclock i
...
-* [AMD](https://i.imgur.com/Ie4LVtI.png)
-* RAM timings are split into 3 categories: primary, secondary and tertiary. These are indicated by 'P', 'S', and 'T' respectively.
+* Intel
+
+  ![image](https://user-images.githubusercontent.com/69487009/156914502-104b0cbd-bbc4-4c94-a583-426e84f732eb.png)
+
+* AMD
+
+  ![image](https://user-images.githubusercontent.com/69487009/156914518-5510c6a2-6187-41ee-8acd-a8f76a5590c4.png)
+
```
Comment (integralfx) [inline DDR4 OC Guide.md:129]: Why did you change the URLs?

File: DDR4 OC Guide.md [hunk 4]
```diff
@@ -170,42 +183,42 @@ Sometimes the Thaiphoon report won't tell you the IC or it may misidentify the I
...
-   * x16 configurations will have half as many banks and bank groups compared to the traditional x8 configurations which means less performance. See [buildzoid's video](https://www.youtube.com/watch?v=k6SIdxq2yxE) for more information.
-* Density matters when determining how far your ICs can go. For example, 4Gb AFR and 8Gb AFR will not overclock the same despite sharing the same name. The same can be said for Micron Rev. B which exists as both 8Gb and 16Gb. The 16Gb ICs overclock better and are sold as both in 16GB and 8GB capacities despite the DIMMs using 8 chips. The 8GB sticks have their SPD modified and can be found in higher-end Crucial kits (BLM2K8G51C19U4B).
-* As the count of ranks total in a system increases, so does the load on the memory controller. This usually means that more memory ranks will a require higher voltage, especially VCCSA on Intel and SOC voltage on AMD.
+   * x16 configurations will have half as many banks and bank groups compared to the traditional x8 configurations which mean less performance. See [buildzoid's video](https://www.youtube.com/watch?v=k6SIdxq2yxE) for more information.
```
Comment (integralfx) [inline DDR4 OC Guide.md]: Pretty sure 'means' is the correct word here

File: DDR4 OC Guide.md [hunk 5]
```diff
@@ -170,42 +183,42 @@ Sometimes the Thaiphoon report won't tell you the IC or it may misidentify the I
...
+* Density matters when determining how far your ICs can go. For example, 4Gb AFR and 8Gb AFR will not overclock the same despite sharing the same name. The same can be said for Micron Rev. B which exists as both 8Gb and 16Gb. The 16Gb ICs overclock better and are sold both in 16GB and 8GB capacities despite the DIMMs using 8 chips. The 8GB sticks have their SPD modified and can be found in higher-end Crucial kits (BLM2K8G51C19U4B).
+* As the count of ranks' total in a system increases, so does the load on the memory controller. This usually means that more memory ranks will require higher voltage, especially VCCSA on Intel and SOC voltage on AMD.
```
Comment (integralfx) [inline DDR4 OC Guide.md]:
```suggestion
* As the total count of ranks in a system increases, so does the load on the memory controller. This usually means that more memory ranks will require higher voltage, especially VCCSA on Intel and SOC voltage on AMD.
```

I think this makes more sense

File: DDR4 OC Guide.md [hunk 6]
```diff
@@ -251,7 +264,7 @@ As far as I know, tCL, tRCD, tRP and possibly tRFC can (or can not) see voltage
 * Binning is basically grading components based on their performance characteristics.  
   Manufacturers would separate ICs into different containers/bins depending on their frequency. Hence the term binning.
 * G.Skill is one manufacturer known for extensive binning and categorization. Oftentimes, several different SKUs of G.Skill memory will belong to the same factory bin (i.e. DDR4-3600 16-16-16-36 1.35V bin of B-Die being binned the same as DDR4-3200 14-14-14-34 1.35V B-Die).
-* B-die binned for 2400 15-15-15 is significantly worse than good B-die binned for DDR4-3200 14-14-14 or even DDR4-3000 14-14-14. Don't expect it to have the same voltage scaling characteristics as good B-Die.
+* B-die binned for 2400 15-15-15 is significantly worse than good B-die binned for DDR4-3200 14-14-14 or even DDR4-3000 14-14-14. Don't expect it to have the same voltage scaling characteristics as a good B-Die.
```
Comment (integralfx) [inline DDR4 OC Guide.md]: It should be either

> as good B-Die

or

> as a good B-Die kit

File: DDR4 OC Guide.md [hunk 7]
```diff
@@ -263,28 +276,28 @@ As far as I know, tCL, tRCD, tRP and possibly tRFC can (or can not) see voltage
...
-* The tRFC timings are very dependent on temperatures, as they are related to capacitor leakage which is affected by temperature. Higher temperatures will need higher tRFC values. tRFC2 and tRFC4 are timings that activate when operating temperature of DRAM hits 85°C. Below these temperatures, these timings don't do anything.
+* The tRFC timings are very dependent on temperatures, as they are related to capacitor leakage which is affected by temperature. Higher temperatures will need higher tRFC values. tRFC2 and tRFC4 are timings that activate when the operating temperature of DRAM hits 85°C. Below these temperatures, these timings don't do anything.
 * B-Die is temperature sensitive and its ideal range is ~30-40°C. Some may be able to withstand higher temperatures so YMMV.
-* Rev. E, on the other hand, doesn't seem to be as strongly temperature sensitive, demonstrated by [buildzoid](https://www.youtube.com/watch?v=OeHEtULQg3Q).
+* Rev. E, on the other hand, doesn't seem to be as strongly temperature-sensitive, demonstrated by [buildzoid](https://www.youtube.com/watch?v=OeHEtULQg3Q).
```
Comment (integralfx) [inline DDR4 OC Guide.md]: I don't think the hyphen is necessary here

File: DDR4 OC Guide.md [hunk 8]
```diff
@@ -302,33 +318,33 @@ As far as I know, tCL, tRCD, tRP and possibly tRFC can (or can not) see voltage
...
+  > There are clear differences in how the memory controller behaves on the different CPU specimens. The majority of the CPUs will do DDR4-3466 or higher at 1.050V SoC voltage, however, the difference lies in how the different specimens react to the voltage. Some of the specimens seem to scale with the increased SoC voltage, while the others simply refuse to scale at all or in some cases even illustrate negative scaling. All of the tested samples illustrated negative scaling (i.e. more errors or failures to train) when higher than 1.150V SoC was used. In all cases, the maximum memory frequency was achieved at =< 1.100V SoC voltage.
```
Comment (integralfx) [inline DDR4 OC Guide.md]: This is a quote so it's not necessary to fix the grammar mistakes

File: DDR4 OC Guide.md [hunk 9]
```diff
@@ -302,33 +318,33 @@ As far as I know, tCL, tRCD, tRP and possibly tRFC can (or can not) see voltage
...
+Because of this, there are a couple of rules. For example, if you set the VDDG to 1.100V, while your actual SoC voltage under load is 1.05V the VDDG will stay roughly at 1.01V max.
+Likewise, if you have VDDG set to 1.100V and start increasing the SoC voltage, your VDDG will raise as well. I don't have the exact figure, but you can assume that the minimum drop-out voltage (Vin-Vout) is around 40mV.
+Meaning your ACTUAL SoC voltage has to be at least by this much higher, than the requested VDDG for it to take effect as it is requested.  
 Adjusting the SoC voltage alone, unlike on previous gen. parts doesn't do much if anything at all.
-The default value is fixed 1.100V and AMD recommends keeping it at that level. Increasing the VDDG helps with the fabric overclocking in certain scenarios, but not always.
+The default value is fixed at .1.100V and AMD recommends keeping it at that level. Increasing the VDDG helps with the fabric overclocking in certain scenarios, but not always.
 1800MHz FCLK should be doable at the default 0.9500V value and for pushing the limits it might be beneficial to increase it to =< 1.05V (1.100 - 1.125V SoC, depending on the load-line).
```
Comment (integralfx) [inline DDR4 OC Guide.md]: This is a quote so it's not necessary to fix the grammar mistakes

File: DDR4 OC Guide.md [hunk 10]
```diff
@@ -382,54 +398,63 @@ The default value is fixed 1.100V and AMD recommends keeping it at that level. I
   * Tighten (lower) timings.
 
 ## Finding a Baseline
-1. On Intel, start off with 1.15V VCCSA and VCCIO.  
-   On AMD, start off with 1.10V SOC.
+1. * Ensure your sticks are in the recommended DIMM slots (usually 2 and 4).
+   * Make sure your CPU overclock is disabled when tuning RAM, as an unstable CPU can lead to memory errors. When pushing high frequency with tight timings, your CPU may become unstable and may need to be re-done.
+   * Make sure your UEFI/Bios is up to date.
```
Comment (integralfx) [inline DDR4 OC Guide.md]:
```suggestion
   * Make sure your UEFI/BIOS is up to date.
```

File: DDR4 OC Guide.md [hunk 11]
```diff
@@ -382,54 +398,63 @@ The default value is fixed 1.100V and AMD recommends keeping it at that level. I
...
    
 ## Trying Higher Frequencies
-* This section is applicable if you're not at the limit of your motherboard, ICs and IMC.  
+* This section is applicable if you're not at the limit of your motherboard, ICs, and IMC.  
   This section is not for those who are having trouble stabilising frequencies within the expected range.
-     * Note that some boards have auto rules that can stifle your progress, an example being tCWL = tCL - 1 which can lead to uneven values of tCWL. Reading the [Miscellaneous Tips](#miscellaneous-tips) might give you insight into your particular platform and your motherboards functionality.
+     * Note that some boards have auto rules that can stifle your progress, an example being tCWL = tCL - 1 which can lead to uneven values of tCWL. Reading the [Miscellaneous Tips](#miscellaneous-tips) might give you insight into your particular platform and your motherboard's functionality.
 1. Intel:
-   * Increase VCCSA and VCCIO to 1.25V.
+   * Increase VCCSA and VCCIO to 1.25V. For ADL, VCCIO does not exist.
```
Comment (integralfx) [inline DDR4 OC Guide.md]: If you're going to mention ADL, add all the necessary info about it (gear 1, IMC voltages and their daily maximums, etc.)

**Review by integralfx — CHANGES_REQUESTED** [review]

File: DDR4 OC Guide.md [hunk 12]
```diff
@@ -37,30 +37,28 @@
...
-  * If you experience issues with all threads crashing upon launch with the extreme config it might help to edit the row "Testing Window Size (Mb)=1408". Replace the window size with your total RAM (minus some margin for Windows) divided by your processors available threads (e.g. 12800/16 = 800 MB per thread).
-* [OCCT](https://www.ocbase.com/) with the dedicated memory test using either SSE or AVX instructions.
-  * Note that AVX and SSE can vary in error detection speed. On Intel-based systems, SSE appears better for testing IMC voltages while AVX appears better for DRAM voltage.
-  * The Large AVX2 CPU test is a great stability test for your CPU and RAM at the same time. The more you tune your ram the harder it'll be to stable in this test.
+  * [Absolut](https://github.com/integralfx/MemTestHelper/blob/oc-guide/absolutnew.cfg)
+  * [PCBdestroyer](https://github.com/integralfx/MemTestHelper/blob/oc-guide/PCBdestroyer.cfg)
+  * [LMHz Universal 2 config](https://github.com/integralfx/MemTestHelper/blob/oc-guide/Universal-2%40LMhz.cfg)
```
Comment (integralfx) [inline DDR4 OC Guide.md]: These configs don't exist in the repo

You need to add it to your repo and use a relative link
e.g. `TM5/extreme%40anta777.cfg`

File: DDR4 OC Guide.md [hunk 13]
```diff
@@ -375,62 +397,82 @@ The default value is fixed 1.100V and AMD recommends keeping it at that level. I
...
+   
+   | IC                                | Daily voltage (V) | Extreme voltage |
+   | :---:                             | :---------:       | :-------------: |
+   | 8Gbit Samsung B-Die               | 1.45 - 1.55       | 1.55 - 1.65     |
+   | 8Gbit Hynix DJR/16Gbit AJR        | 1.45 - 1.55       | 1.50 - 1.65     |
+   | 8Gbit Micron Rev. E/16Gbit Rev. B | 1.45 - 1.55       | 1.55 - 1.60     |
+   | 8Gbit Hynix CJR                   | 1.35 - 1.40       | 1.40 - 1.45     |
+   | 16Gbit Hynix CJR                  | 1.40 - 1.50       | 1.50 - 1.55     |
+   | 8Gbit Nanya B-Die                 | 1.35 - 1.40       | 1.40 - 1.45     |
+   | Samsung C-Die                     | 1.30 - 1.35       | Likely Unsafe   |
```
Comment (integralfx) [inline DDR4 OC Guide.md]: This should go under Maximum Recommended Daily Voltage

Clarify what extreme voltage means

**Review by integralfx — CHANGES_REQUESTED** [review]

**Comment by integralfx** [comment]
**It's up to the starter of the conversation to resolve the conversation.** 

You should leave a comment saying that you agree/disagree. If you disagree, provide a reason.

Don't just close the conversation without saying anything. That's just a huge middle finger to the starter of the conversation, especially if you keep repeatedly doing this (without even addressing the issue).

File: DDR4 OC Guide.md (hunk 4 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: I agree.

File: DDR4 OC Guide.md (hunk 5 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: I agree, it has been added.

File: DDR4 OC Guide.md (hunk 6 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: This has been fixed.

File: DDR4 OC Guide.md (hunk 7 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: I realize that is correct.

File: DDR4 OC Guide.md (hunk 8 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: It has been fixed.

File: DDR4 OC Guide.md (hunk 9 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: Fixed.

File: DDR4 OC Guide.md (hunk 10 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: This suggestion has been committed.

File: DDR4 OC Guide.md (hunk 12 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: I was thinking of the files being added to your repo directly but that may be too messy with all the files. I'll go do that.

Comment (integralfx) [reply DDR4 OC Guide.md]: You can't add the files directly to my repo anyway.

You need to add them to yours so that the changes get merged into my branch.

I'd recommend you read into git branching: https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow

File: DDR4 OC Guide.md (hunk 13 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: That makes sense, I will also add a link to the section in the part where they overclock.

File: DDR4 OC Guide.md (hunk 11 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: Information about those has been added, as well as daily maximums are not fully known due to it being new.

File: DDR4 OC Guide.md (hunk 3 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md:129]: This is just to embed the images.

Comment (integralfx) [reply DDR4 OC Guide.md:129]: What do you mean by embed?

Comment (LunarPSD) [reply DDR4 OC Guide.md:129]: People won't have to click the link in order to see the images and it'll appear in the document as an image. You can see this in preview mode.

File: DDR4 OC Guide.md (hunk 13 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: This has been moved and clarified.

File: DDR4 OC Guide.md (hunk 12 above)
Comment (LunarPSD) [reply DDR4 OC Guide.md]: I have added relative links to all the files.

**Review by integralfx — APPROVED** [review]

**Review by integralfx — APPROVED** [review]
